tests/test_telemetry.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+# Helper functions
+def get_telemetry_keyvalues(gnmi_out, certs_out):
+    gnmi_values = gnmi_out.split('\n')
+    cert_values = certs_out.split('\n')
+    gnmi_list = []
+    gnmi_list = gnmi_values + cert_values
+    params_dict= dict(itertools.izip_longest(*[iter(gnmi_list)] *2, fillvalue=""))
+    return params_dict
+
+# Test functions
+def test_config_db_parameters(duthost):
+    """Verifies required telemetry parameters from config_db.
+    """
+    gnmi = duthost.shell('/usr/bin/redis-cli -n 4 hgetall "TELEMETRY|gnmi"', module_ignore_errors=True)['stdout']
+    certs = duthost.shell('/usr/bin/redis-cli -n 4 hgetall "TELEMETRY|certs"', module_ignore_errors=True)['stdout']
+    d = get_telemetry_keyvalues(gnmi, certs)
+    for key, value in d.items():
+        logger.info(' testing {}={}'.format(key,value))
+        if str(key) == "client_auth" and str(value) == "true":
+            assert True, "Client_auth set to true"
+        if str(key) == "port" and str(value) == "50051":
+            assert True, "port is set to 50051"
+        if str(key) == "ca_crt" and str(value) == "/etc/sonic/telemetry/dsmsroot.cer":
+            assert True, "ca_crt is set to {}".format(str(value))
+        if str(key) == "server_key" and str(value) == "/etc/sonic/telemetry/streamingtelemetryserver.key":
+            assert True, "server_key is set to {}".format(str(value))
+        if str(key) == "server_crt" and str(value) == "/etc/sonic/telemetry/streamingtelemetry.cer":
+            assert True, "server_crt is set to {}".format(str(value))
```

Comment:
i think it makes sense to check if those file exists referenced in config db exists or not.

Reply:
Agree but this location will be created and cert copy will be done by Hardware Proxy.

Reply:
I think you need to valid if the service can be started correctly. the test here should be self contained without relying on the hwproxy.

Reply:
Basically cert creation/copy can be done in test case 4 which is an end to end testing of telemetry agent and gnmi client. The scope of this test is just to verify config_db parameters. Let me know your thoughts

Reply:
I had an offline discussion. Creat creation/copy should be part of deploy_minigraph. In pytest we will just verify if they exist at a particular location or not. Also, this is not a blocker for current PR